### PR TITLE
Add AWS SDK

### PR DESCRIPTION
**Pull request by JinheLin**
### What problem does this PR solve?

Issue Number: ref #6827

### What is changed and how it works?
- contrib/aws* submodules reference to official repository and the latest release version, except aws-c-cal since the official repository has some conflict with boringssl.
- Add a configuration of [storage.s3].
- Add some basic code of AWS SDK.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by JaySon-Huang** [comment]
Should also add the S3::ClientFactory and its init https://github.com/pingcap/tiflash/pull/6788/files

File: contrib/aws-cmake/AwsSIMD.cmake [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    check_c_source_compiles("
+        #include <immintrin.h>
+        #include <emmintrin.h>
+        #include <string.h>
+
+        int main() {
+            __m256i vec;
+            memset(&vec, 0, sizeof(vec));
+
```
Comment (Lloyd-Pottiger) [inline contrib/aws-cmake/AwsSIMD.cmake:32]: No need, we enable avx2 by default.  Use `TIFLASH_ENABLE_AVX_SUPPORT` to decide whether should be built with avx2.

**Review by Lloyd-Pottiger — APPROVED** [review]
Others LGTM.

**Comment by JaySon-Huang** [comment]
> > Should also add the S3::ClientFactory and its init https://github.com/pingcap/tiflash/pull/6788/files
> 
> Should we initialize AWS SDK by default in the master branch? @JaySon-Huang

I think we should add a config "storage.remote.s3.bucket", if the user set the bucket name, then init the AWS SDK

**Comment by JaySon-Huang** [comment]
And the access key && secret key should be get from ENV variables:
```
export AWS_ACCESS_KEY_ID=test
export AWS_SECRET_ACCESS_KEY=test
```

**Review by JaySon-Huang — APPROVED** [review]
LGTM

**Pull request merged**